x-pack/plugins/lens/public/xy_visualization/axis_settings_popover.tsx

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, useEffect, useCallback } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiText,
+  EuiSwitch,
+  EuiSpacer,
+  EuiFieldText,
+  IconType,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { LayerConfig, AxesSettingsConfig } from './types';
+import { FramePublicAPI } from '../types';
+import { ToolbarPopover } from '../shared_components';
+import { ToolbarButtonProps } from '../toolbar_button';
+// @ts-ignore
+import { EuiIconAxisBottom } from '../assets/axis_bottom';
+// @ts-ignore
+import { EuiIconAxisLeft } from '../assets/axis_left';
+// @ts-ignore
+import { EuiIconAxisRight } from '../assets/axis_right';
+
+type AxesSettingsConfigKeys = keyof AxesSettingsConfig;
+export interface AxisSettingsPopoverProps {
+  /**
+   * Determines the popover title
+   */
+  popoverTitle: string;
+  /**
+   * Determines the axis
+   */
+  axis: AxesSettingsConfigKeys;
+  /**
+   * Contains the chart layers
+   */
+  layers?: LayerConfig[];
+  /**
+   * Determines the axis title
+   */
+  axisTitle: string | undefined;
+  /**
+   * Callback to axis title change
+   */
+  updateTitleState: (value: string) => void;
+  /**
+   * Determines the frame
+   */
+  frame: FramePublicAPI;
+  /**
+   * Determines if the popover is Disabled
+   */
+  isDisabled?: boolean;
+  /**
+   * Determines if the ticklabels of the axis are visible
+   */
+  areTickLabelsVisible: boolean;
+  /**
+   * Toggles the axis tickLabels visibility
+   */
+  toggleTickLabelsVisibility: (axis: AxesSettingsConfigKeys) => void;
+  /**
+   * Determines if the gridlines of the axis are visible
+   */
+  areGridlinesVisible: boolean;
+  /**
+   * Toggles the gridlines visibility
+   */
+  toggleGridlinesVisibility: (axis: AxesSettingsConfigKeys) => void;
+  /**
+   * Determines if the title visibility switch is on and the input text is disabled
+   */
+  isAxisTitleVisible: boolean;
+  /**
+   * Toggles the axis title visibility
+   */
+  toggleAxisTitleVisibility: (axis: AxesSettingsConfigKeys, checked: boolean) => void;
+}
+const popoverConfig = (
+  axis: AxesSettingsConfigKeys
+): { icon: IconType; groupPosition: ToolbarButtonProps['groupPosition'] } => {
```

Comment:
These icons don't match the axes for horizontal charts. Same with the choice to label them as "left" and "right" axes. It's not even a perfect 90º rotation, because in a horizontal chart the "right" axis is actually rendered on top, and the "left" axis is rendered on bottom.

Reply:
You are totally right, I didn't think about it. @cchaos what do you think about it? We could change the icons and copies in case of horizontal bar but I would also love your feedback here

Reply:
@cchaos I have made it work for horizontal bars but I am missing the TopAxis Icon. Can you help me with it?